Orange/widgets/widget.py

```diff
@@ -197,117 +207,132 @@ def __init__(self, *args, **kwargs):
+            self.controlArea = self.left_side
+        if self.want_main_area:
+            self.controlArea.setSizePolicy(
+                QSizePolicy.Fixed, QSizePolicy.MinimumExpanding)
+            m = 0
+        else:
+            m = 4
+        self.controlArea.layout().setContentsMargins(m, m, m, m)
+
+    def _insert_buttons_area(self):
+        buttons = []
+        if self.graphButton is not None:
+            buttons.append(self.graphButton)
+        if self.report_button is not None:
+            buttons.append(self.report_button)
+        orientation = self.want_buttons_area
+        if orientation == "auto":
+            orientation = "horizontal" if len(buttons) <= 2 else "vertical"
+        self.buttonsArea = gui.widgetBox(
+            self.left_side, addSpace=0, spacing=9, orientation=orientation)
+        for button in buttons:
+            self.buttonsArea.layout().addWidget(button)
+
+    def _insert_main_area(self):
+        self.mainArea = gui.vBox(
+            self.splitter, margin=4,
+            sizePolicy=QSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
+        )
+        self.splitter.setCollapsible(1, False)
+        self.mainArea.layout().setContentsMargins(
+            0 if self.want_control_area else 4, 4, 4, 4)
+
+    def _insert_status_bar(self):
+        self.widgetStatusArea = status_area = QFrame(self)
+        status_area.setLayout(QHBoxLayout(status_area))
+        status_area.layout().setMargin(0)
+        status_area.setFrameShape(QFrame.StyledPanel)
+        self.layout().addWidget(status_area)
+
+        self.statusBarIconArea = QFrame(self)
+        self.statusBarIconArea.setLayout(QHBoxLayout())
+        self.statusBarIconArea.hide()
+        status_area.layout().addWidget(self.statusBarIconArea)
+
+        self.widgetStatusBar = QStatusBar(self)
+        self.widgetStatusBar.setSizeGripEnabled(0)
+        status_area.layout().addWidget(self.widgetStatusBar)
+
+        for attr, name in (("_warningWidget", "icons/triangle-orange.png"),
+                           ("_errorWidget", "icons/triangle-red.png")):
+            w = QLabel(self.statusBarIconArea)
+            self.statusBarIconArea.layout().addWidget(w)
+            w.setFixedSize(16, 16)
+            w.hide()
+            filename = gui.resource_filename(name)
+            if os.path.exists(filename):
+                w.setPixmap(QPixmap(filename))
+            setattr(self, attr, w)
+
+    def _create_default_buttons(self):
+        # These button is inserted in buttons_area, if it exists
+        # Otherwise it is up to the widget to add them to some layout
+        if self.graph_name is not None:
+            self.graphButton = QPushButton("&Save Image", autoDefault=False)
+            self.graphButton.clicked.connect(self.save_graph)
+        if hasattr(self, "send_report"):
+            self.report_button = QPushButton("&Report", autoDefault=False)
+            self.report_button.clicked.connect(self.show_report)
```

Comment:
No. I extracted this from `_insert_buttons_area` so the buttons can be created without also creating `buttonsArea`.